### ISSUE TITLE
Incorrect reporting of unused bindings after 4th function in letfn binding

### ISSUE THREAD

**Issue by vincentjames501** [clj-kondo/clj-kondo#2386]
- [x ] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

clj-kondo v2024.08.29

**platform**

Native version (MacOS)

**problem**

clj-kondo incorrectly reports unused bindings after 4th function in letfn binding

```
src/cljkondolint.clj:12:18: warning: unused binding a
src/cljkondolint.clj:12:20: warning: unused binding b
src/cljkondolint.clj:14:18: warning: unused binding a
src/cljkondolint.clj:14:20: warning: unused binding b
```

**repro**

```
(def xxx
  (letfn [(abc [a b]
            (+ a b))
          (abc2 [a b]
            (+ a b))
          (abc3 [a b]
            (+ a b))
          (abc4 [a b]
            (+ a b))
          (abc5 [a b]
            (+ a b))
          (abc6 [a b]
            (+ a b))]
    (abc 1 2)
    (abc2 1 2)
    (abc3 1 2)
    (abc4 1 2)
    (abc5 1 2)
    (abc6 1 2)))
```

**Comment by borkdude** [comment]
Thanks. Did this problem also occur with the previous version?

**Comment by vincentjames501** [comment]
@borkdude , ah I forgot to say! No it did not. I did a brew update and started getting this. I'm not super sure exactly which version I came from but I believe it must be pretty recent.

**Comment by borkdude** [comment]
Found the issue. Bitten by Clojure laziness. I'll merge the fix tomorrow.
